hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileSystemContract.java

```diff
@@ -137,4 +142,30 @@ public void testRenameNonExistentPath() throws Exception {
         () -> super.testRenameNonExistentPath());
 
   }
+
+  @Test
+  public void testFileSystemCapabilities() throws Exception {
```

Comment:
if the methods are on an interface only implemented by a few stores, there's no need to worry about s3a, abfs etc